internal/drivers/ankabuild/driver.go

```diff
@@ -104,8 +106,15 @@ func (c *config) Create(ctx context.Context, opts *types.InstanceCreateOpts) (in
 		deleteErr := c.ankaClient.VMDelete(ctx, vm.Body.InstanceID)
 		if deleteErr != nil {
 			logrus.Errorf("ankabuild: error deleting vm: %s", deleteErr)
+			return nil, deleteErr
+		}
+		// VMs creation can randomly fail, retry 3 times before returning error
+		if retry < 3 {
+			logrus.Infof("ankabuild: retry vm creation attempt: %d", retry)
+			retry++
```

Comment:
wouldn't this behaviour be different from expected? You have retry as a global variable so anytime Create is called it will increment retry by 1 - for each Create() from any instance of config.
Also, it might be better to retry the specific API call which needs to be retried as opposed to calling the function again

Reply:
Ready for another review